extensions/wikia/AdEngine/js/video/uapVideo.js

```diff
@@ -29,17 +39,16 @@ define('ext.wikia.adEngine.video.uapVideo', [
 
 		return porvata.inject(params)
 			.then(function (video) {
-				videoInterface.setup(video, [
-					'progressBar',
-					'pauseOverlay',
-					'volumeControl',
-					'closeButton',
-					'toggleAnimation'
-				], {
+				var uiElements = params.autoPlay ? UITemplate.autoPlay : UITemplate.default;
+
+				log(['VUAP UI elements', uiElements], log.levels.debug, logGroup);
+
+				videoInterface.setup(video, uiElements, {
```

Comment:
videoInterface only works with porvata video. How about defining method called `setupUI` on porvata player instance. This instance should have knowledge about itself - maybe we can avoid passing `params` object through 4 files.

However, above change may be tricky because we don't want to set unnecessary params on porvata player - when deciding which params should be defined as part of porvata player instance keep in mind what is the player responsibility.